simple_slice_cython.pyx

```diff
@@ -0,0 +1,131 @@
+# distutils: language = c
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: initializedcheck=False
+
+from cpython cimport PyObject
+from libc.stdint cimport int64_t
+import sys
+
+cdef extern from "Python.h":
+    int PyIndex_Check(object obj)
+    object PyNumber_Index(object obj)
+    bint PyBool_Check(object obj)
+    int64_t PyLong_AsLongLong(object obj) except? -1
+
+cdef bint _NUMPY_IMPORTED = False
+cdef type _NUMPY_BOOL = None
+
+cdef inline bint is_numpy_bool(object obj):
+    global _NUMPY_IMPORTED, _NUMPY_BOOL
+    if not _NUMPY_IMPORTED:
+        if 'numpy' in sys.modules:
+            _NUMPY_BOOL = sys.modules['numpy'].bool_
+        _NUMPY_IMPORTED = True
+    return _NUMPY_BOOL is not None and isinstance(obj, _NUMPY_BOOL)
+
+cdef inline int64_t cy_operator_index(object idx) except? -1:
+    cdef object result
+
+    if PyBool_Check(idx) or is_numpy_bool(idx):
+        raise TypeError(f"'{type(idx).__name__}' object cannot be interpreted as an integer")
+
+    if PyIndex_Check(idx):
```

Comment:
Maybe this `if PyIndex_Check(idx)` and the following logic can re replaced by a single call to  `PyNumber_AsSsize_t` (cf https://docs.python.org/3/c-api/number.html): 

> Returns o converted to a [Py_ssize_t](https://docs.python.org/3/c-api/intro.html#c.Py_ssize_t) value if o can be interpreted as an integer. If the call fails, an exception is raised and -1 is returned.